torch/csrc/jit/passes/onnx.cpp

```diff
@@ -186,11 +187,40 @@ void BlockToONNX(
     return it->second;
   };
 
+  GRAPH_DEBUG("BlockToONNX: graph of old block: ", old_block->owningGraph()->toString());
+
   // Initialize context and environment
   for (auto input : old_block->inputs()) {
     auto n = ctx.block->addInput()->copyMetadata(input);
     env[input] = n;
   }
+
+  // Merge ONNX inferred type with existing type.
```

Comment:
I think this comments needs to fleshed out more, to explain the logic of merge in detail. It will help for posterity. For example, what does "merge the two types" mean. Someone who is reading this code for the first time should be able to understand what is supposed to happen in this method and how.

Reply:
Updated the comments, and moved this function to `shape_type_inference.cpp` file, since it deals with inferred types and type merging logics